bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/tracer/TracersStore.java

```diff
@@ -55,25 +58,32 @@ public void setTracerGenerator(TracerProvider tracerProvider) {
      * @return trace implementations i.e: zipkin, jaeger
      */
     public Tracer getTracer(String serviceName) {
-        Tracer openTracer;
+        Tracer tracer;
         if (store.containsKey(serviceName)) {
-            openTracer = store.get(serviceName);
+            tracer = store.get(serviceName);
         } else {
             if (tracerProvider != null) {
                 try {
-                    openTracer = tracerProvider.getTracer(serviceName);
+                    tracer = tracerProvider.getTracer(serviceName);
                 } catch (Throwable e) {
-                    openTracer = NoopTracerFactory.create();
+                    tracer = io.opentelemetry.api.trace.TracerProvider.noop().get("");
                     consoleError.println("error: tracing disabled as getting tracer for " + serviceName + " service. "
                             + e.getMessage());
                 }
-                store.put(serviceName, openTracer);
+                store.put(serviceName, tracer);
             } else {
-                openTracer = NoopTracerFactory.create();
+                tracer = io.opentelemetry.api.trace.TracerProvider.noop().get("");
                 consoleError.println("error: tracing disabled as tracer provider had not been initialized");
             }
         }
-        return openTracer;
+        return tracer;
+    }
+
+    public ContextPropagators getPropagators() {
+        if (propagators != null) {
+            return propagators;
+        }
+        return ContextPropagators.noop();
```

Comment:
Shall we set `propagators` field to `ContextPropagators.noop()` within the `setTracerGenerator` method ?

Reply:
Then we can remove this logic I think.

Reply:
Then `getPropagators()` method will return null if its called before `setTracerGenerator()` method. That is different to the logic in `getTracer()` method.
`tracerProvider` being null case is handled in `getTracer` method, I wrote a similar logic in `getPropagators` method.